src/routes/ComparisonCreateRoute.js

```diff
@@ -166,6 +167,21 @@ class ComparisonCreateRoute extends React.Component {
 
         history.push(`/comparisons-erm/${comparisonId}${location.search}`);
         this.context.sendCallout({ message: <SafeHTMLMessage id="ui-erm-comparisons.comparison.created.success" values={{ name }} /> });
+      })
+      .catch(response => {
+        response.json()
+          .then(error => {
+            if (error.errors.some(err => (
+              err?.message.includes('Property [titleList] of class [class org.olf.erm.ComparisonPoint] cannot be null')))
+            ) {
+              this.context.sendCallout({ type: 'error', message: <SafeHTMLMessage id="ui-erm-comparisons.comparison.created.error.noComparisonLinked" /> });
+            } else {
+              error.errors.forEach(err => (
+                this.context.sendCallout({ type: 'error', message: <SafeHTMLMessage id="ui-erm-comparisons.comparison.created.error.generalError" values={{ err: err.message }} /> })
+              ));
+            }
+          })
+          .catch(() => this.context.sendCallout({ type: 'error', message: <SafeHTMLMessage id="ui-erm-comparisons.comparison.created.error.unknownError" /> }));
```

Comment:
Can we have this entire code block be more dynamic? something similar to https://github.com/folio-org/ui-local-kb-admin/blob/master/src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesForm.js#L21

Reply:
I've made a change here, but I'm not sure it's quite what you were after. Let me know what you think